stdlib/Dates/src/ranges.jl

```diff
@@ -15,12 +15,13 @@ function len(a, b, c)
     i = guess(a, b, c)
     v = lo + st * i
     prev = v  # Ensure `v` does not overflow
+    hi -= st # Prevent overshooting
     while v <= hi && prev <= v
         prev = v
         v += st
         i += 1
     end
-    return i - 1
+    return i
```

Comment:
ah, this is still a work in progress to figure out how to do this. The range implementations currently rely on being able to go past the actual end of the supported range, and now that that errors, this needs to be rethought

Reply:
Perhaps it would help to extend the constructable dates from `typemin(Date):typemax(Date)` to the slightly larger set of representable dates `typemin(Date)-Day(161):typemax(Date)+Day(280)`